### PR TITLE
Add missing methods

### PR DESCRIPTION
**Pull request by vincentsarago**
closes https://github.com/developmentseed/morecantile/issues/48

add `parent`, `children`, `neighbors`, `extrema`, `is_valid` methods

There are some difference with how we get the parent/children in comparison to mercantile method because we can't really on only the 1 <-> 4 schema of WebMercator. The general idea is to get the bounds from the tile and then find tile(s) for the bounds at the given zoom level. Tests give the same result as for mercantile 🤷‍♂️

### CODE REVIEW

File: morecantile/models.py [hunk 1]
```diff
@@ -925,3 +930,154 @@ def quadkey_to_tile(self, qk: str) -> Tile:
...
+
+        res = self._resolution(self.matrix(t.z)) / 10.0
+
+        bbox = self.xy_bounds(t)
+        ul_tile = self._tile(bbox.left + res, bbox.top - res, target_zoom)
+        lr_tile = self._tile(bbox.right - res, bbox.bottom + res, target_zoom)
+
+        tiles = []
+        for i in range(ul_tile.x, lr_tile.x + 1):
+            for j in range(ul_tile.y, lr_tile.y + 1):
```
Comment (vincentsarago) [inline morecantile/models.py:1042]: simple `parent` finding: get bounds of the tile and find tiles for the bounds at zoom - 1 🤷‍♂️ 

big diff with mercantile is that we return a list of tile, that's said I'm not 💯 sure that we could have multiple parents 🤷‍♂️

File: morecantile/models.py [hunk 2]
```diff
@@ -925,3 +930,154 @@ def quadkey_to_tile(self, qk: str) -> Tile:
...
+
+        target_zoom = t.z + 1 if zoom is None else zoom
+
+        bbox = self.xy_bounds(t)
+        res = self._resolution(self.matrix(t.z)) / 10.0
+
+        ul_tile = self._tile(bbox.left + res, bbox.top - res, target_zoom)
+        lr_tile = self._tile(bbox.right - res, bbox.bottom + res, target_zoom)
+
+        tiles = []
```
Comment (vincentsarago) [inline morecantile/models.py:1078]: Same as for `parent`: get tile bounds and find tile for zoom + 1

File: morecantile/models.py [hunk 3]
```diff
@@ -925,3 +930,154 @@ def quadkey_to_tile(self, qk: str) -> Tile:
...
+        Returns
+        -------
+        Dict
+
+        """
+        m = self.matrix(zoom)
+        return {
+            "x": {"min": 0, "max": m.matrixWidth - 1},
+            "y": {"min": 0, "max": m.matrixHeight - 1},
+        }
```
Comment (vincentsarago) [inline morecantile/models.py:953]: this is like the `minmax` method of mercantile but some TMS can have diff min/max for each axis

File: morecantile/models.py (hunk 1 above)
Comment (geospatial-jeff) [reply morecantile/models.py:1042]: I think you technically can have multiple parents if the TMS doesn't decimate consistently, or has different tile width/heights or top left corner at different zooms.  I'm not sure how often you'd run into this in the wild though.

File: morecantile/models.py [hunk 4]
```diff
@@ -925,3 +930,154 @@ def quadkey_to_tile(self, qk: str) -> Tile:
                 raise QuadKeyError("Unexpected quadkey digit: %r", digit)
 
         return Tile(xtile, ytile, i + 1)
+
+    def extrema(self, zoom: int) -> Dict:
```
Comment (geospatial-jeff) [inline morecantile/models.py:934]: should this be called `def minmax` to match mercantile?

**Review by geospatial-jeff — APPROVED** [review]

**Pull request merged**